src/File.php

```diff
@@ -485,6 +486,27 @@ public function canDelete($member = null)
             ->canDelete($this->ID, $member);
     }
 
+    /**
+     * Check if the File is viewable to a public user i.e. someone who is not logged in to the CMS
+     */
+    public function canViewAnonymous(): bool
+    {
+        return Member::actAs(null, function(): bool {
+            return $this->canView();
+        });
+    }
+
+    /**
+     * Whether the file was originally uploaded in a userform submissin on a FileFiled
+     */
+    public function isUserDefinedFormUpload(): bool
+    {
+        if (!class_exists('SilverStripe\\UserForms\\Model\\Submission\\SubmittedFileField')) {
```

Comment:
can use either `\` or `\\` here, single looks prettier and is a bit more standard throughout the rest of the modules

Reply:
`class_exists(SubmittedFileField::class)` will work too if you’ve already imported the class name 😉

Reply:
yeah though what if you haven't actually imported it :-) The `use` statement to import the class will do nothing and not error if the class does not exist.  It'll be pretty normal for an install to not have user-defined-forms

Reply:
Yep but PHP will magically resolve it to a string, so `class_exists()` will work without erroring 😄 https://3v4l.org/gf3kr

Reply:
well ok, learn something everyday :-)